soil-texture-hsg fails when more layers defined

```sh
wget --post-file ows/wps/tests/request-soil-texture-hsg.xml "https://rain1.fsv.cvut.cz/services/wps?" 
--2021-10-15 13:39:29--  https://rain1.fsv.cvut.cz/services/wps?
Resolving rain1.fsv.cvut.cz (rain1.fsv.cvut.cz)... 147.32.140.150
Connecting to rain1.fsv.cvut.cz (rain1.fsv.cvut.cz)|147.32.140.150|:443... connected.
HTTP request sent, awaiting response... 400 BAD REQUE
...
```

PyWPS log says:

```
2021-10-15 11:41:02,675] [ERROR] file=/usr/local/lib/python3.6/dist-packages/pywps/exceptions.py line=45 module=exceptions function=__init__ Exception: code: 400, description: Input data not valid using mode 1, locator: 
NoneType: None
```

Probably related to https://github.com/ctu-geoforall-lab/rain/commit/eb21cdf4e14e3145810f3fea1c72e96d0d848575

![Screenshot from 2021-10-15 13-38-12](https://user-images.githubusercontent.com/5683186/137481442-0691180b-f2f9-4fd9-b610-cdf00653b964.png)

QGIS plugin still fails: 

```
2021-11-09 17:27:04,799] [ERROR] file=/usr/local/lib/python3.6/dist-packages/pywps/exceptions.py line=45 module=exceptions function=__init__ Exception: code: 400, description: Input data not valid using mode 1, locator: 
NoneType: None
```

![Screenshot from 2021-11-09 18-20-59](https://user-images.githubusercontent.com/5683186/140974221-936544cd-1494-480c-a0ec-f473470652af.png)

--- a/ows/wps/pywps/processes/soil_texture_hsg.py
+++ b/ows/wps/pywps/processes/soil_texture_hsg.py
@@ -55,7 +55,7 @@ class SoilTextureHsgProcess(Process):
         
     def _handler(self, request, response):
         # check layers
-        layers = [l.data for l in request.inputs['layers']]
+        layers = [l.data.strip() for l in request.inputs['layers']]
         for lyr in layers:
             if lyr not in self.layers:
                 raise ProcessError("Neplatna vrstva: {}".format(lyr))

--- a/ows/wps/pywps/processes/soil_texture_hsg.py
+++ b/ows/wps/pywps/processes/soil_texture_hsg.py
@@ -22,8 +22,9 @@ class SoilTextureHsgProcess(Process):
             LiteralInput(
                 identifier="layers",
                 title="Vrstvy hydropedologickych charakteristik ({})".format(",".join(self.layers)),
-                data_type='string'
-                #               , allowed_values=self.layers
+                data_type='string',
+                allowed_values=self.layers,
+                max_occurs=4
             )
         ]
         outputs = [
@@ -37,7 +38,7 @@ class SoilTextureHsgProcess(Process):
 
         super(SoilTextureHsgProcess, self).__init__(
             self._handler,
-            identifier="soil-texture-hsg",
+            identifier="soil-texture-hsg-issue",
             version="1.0",
             title="Zrnitost a hydrologicka skupina pudy",
             abstract="Sluzba vraci vyrez rastru zvolene hydropedologicke charakteristiky dle zadaneho polygonu o velikosti do 20 km2.",
@@ -54,7 +55,7 @@ class SoilTextureHsgProcess(Process):
         
     def _handler(self, request, response):
         # check layers
-        layers = request.inputs['layers'][0].data.split(",")
+        layers = [l.data for l in request.inputs['layers']]
         for lyr in layers:
             if lyr not in self.layers:
                 raise ProcessError("Neplatna vrstva: {}".format(lyr))

--- a/ows/wps/pywps/processes/soil_texture_hsg.py
+++ b/ows/wps/pywps/processes/soil_texture_hsg.py
@@ -38,7 +38,7 @@ class SoilTextureHsgProcess(Process):
 
         super(SoilTextureHsgProcess, self).__init__(
             self._handler,
-            identifier="soil-texture-hsg-issue",
+            identifier="soil-texture-hsg",
             version="1.0",
             title="Zrnitost a hydrologicka skupina pudy",
             abstract="Sluzba vraci vyrez rastru zvolene hydropedologicke charakteristiky dle zadaneho polygonu o velikosti do 20 km2.",

--- a/ows_tests/startup.sh
+++ b/ows_tests/startup.sh
@@ -7,7 +7,7 @@ while true; do
     LOGFILE=/var/log/ows_tests/rain_ows_tests-`date -Is`.log
     RETCODE=0
     echo "============================= WPS TESTS ==============================" > $LOGFILE
-    python3 -m pytest -v -s -o cache_dir=/tmp/pytest_cache_dir ./tests/wps/test_wps.py >> $LOGFILE
+    python3 -m pytest -v -o cache_dir=/tmp/pytest_cache_dir ./tests/wps/test_wps.py >> $LOGFILE
     RETCODE=$((RETCODE+$?))
     python notifier.py $RETCODE $LOGFILE
     echo "Sleeping for ${OWS_TEST_REPEAT}hrs..."

--- a/ows_tests/wps/request-soil-texture-hsg.xml
+++ b/ows_tests/wps/request-soil-texture-hsg.xml
@@ -38,15 +38,26 @@
     <wps:Input>
       <ows:Identifier>layers</ows:Identifier>
       <wps:Data>
-	<wps:LiteralData>sand,silt,clay,usda-texture-class</wps:LiteralData>
+	<wps:LiteralData>sand</wps:LiteralData>
+      </wps:Data>
+    </wps:Input>
+    <wps:Input>
+      <ows:Identifier>layers</ows:Identifier>
+      <wps:Data>
+	<wps:LiteralData>silt</wps:LiteralData>
+      </wps:Data>
+    </wps:Input>
+    <wps:Input>
+      <ows:Identifier>layers</ows:Identifier>
+      <wps:Data>
+	<wps:LiteralData>clay</wps:LiteralData>
+      </wps:Data>
+    </wps:Input>
+    <wps:Input>
+      <ows:Identifier>layers</ows:Identifier>
+      <wps:Data>
+	<wps:LiteralData>usda-texture-class</wps:LiteralData>
       </wps:Data>
     </wps:Input>
   </wps:DataInputs>
-  <!-- <wps:ResponseForm> -->
-  <!--   <wps:ResponseDocument lineage="false" storeExecuteResponse="false" status="false"> -->
-  <!--     <wps:Output asreference="false" mimeType="application/x-zipped-shp"> -->
-  <!--       <ows:Identifier>output</ows:Identifier> -->
-  <!--     </wps:Output> -->
-  <!--   </wps:ResponseDocument> -->
-  <!-- </wps:ResponseForm> -->
 </wps:Execute>

--- a/ows_tests/wps/request-soil-texture-hsg20km2.xml
+++ b/ows_tests/wps/request-soil-texture-hsg20km2.xml
@@ -38,15 +38,26 @@
     <wps:Input>
       <ows:Identifier>layers</ows:Identifier>
       <wps:Data>
-	<wps:LiteralData>sand,silt</wps:LiteralData>
+	<wps:LiteralData>sand</wps:LiteralData>
+      </wps:Data>
+    </wps:Input>
+    <wps:Input>
+      <ows:Identifier>layers</ows:Identifier>
+      <wps:Data>
+	<wps:LiteralData>silt</wps:LiteralData>
+      </wps:Data>
+    </wps:Input>
+    <wps:Input>
+      <ows:Identifier>layers</ows:Identifier>
+      <wps:Data>
+	<wps:LiteralData>clay</wps:LiteralData>
+      </wps:Data>
+    </wps:Input>
+    <wps:Input>
+      <ows:Identifier>layers</ows:Identifier>
+      <wps:Data>
+	<wps:LiteralData>usda-texture-class</wps:LiteralData>
       </wps:Data>
     </wps:Input>
   </wps:DataInputs>
-  <!-- <wps:ResponseForm> -->
-  <!--   <wps:ResponseDocument lineage="false" storeExecuteResponse="false" status="false"> -->
-  <!--     <wps:Output asreference="false" mimeType="application/x-zipped-shp"> -->
-  <!--       <ows:Identifier>output</ows:Identifier> -->
-  <!--     </wps:Output> -->
-  <!--   </wps:ResponseDocument> -->
-  <!-- </wps:ResponseForm> -->
 </wps:Execute>

--- a/ows_tests/wps/test_wps.py
+++ b/ows_tests/wps/test_wps.py
@@ -7,8 +7,8 @@ from owslib.wps import WebProcessingService, monitorExecution, ComplexDataInput
 from osgeo import ogr, gdal
 
 class TestWPS:
-    url='https://rain1.fsv.cvut.cz/services/wps'
-    # url='http://localhost/services/wps'
+    # url='https://rain1.fsv.cvut.cz/services/wps'
+    url='http://localhost/services/wps'
     input_data=ComplexDataInput("http://rain.fsv.cvut.cz/geodata/test.gml")
     key="HLGP_ID"
     return_period="N2,N5,N100"
@@ -42,16 +42,24 @@ class TestWPS:
 
         return ofile
 
-    def _run_job_request(self, request_file, ext, url=None):
+    def _run_job_request(self, request_file, ext=None, url=None,
+                         exception=None):
         with open(request_file,'rb') as fd:
             request = fd.read()
         execution = self._wps(url).execute(None, [], request=request)
         monitorExecution(execution)
-        assert execution.getStatus() == "ProcessSucceeded"
-        ofile = self._get_filename() + ext
-        execution.getOutput(ofile)
 
-        return ofile
+        if exception is None:
+            assert execution.getStatus() == "ProcessSucceeded"
+            ofile = self._get_filename() + ext
+            execution.getOutput(ofile)
+            return ofile
+        else:
+            assert len(execution.errors) > 0
+            assert execution.getStatus() == "Exception"
+            assert str(execution.errors[0].text).startswith(exception)
+
+        return None
     
     def _process_d_rain_output(self, ofile, units=False):
         if Path(ofile).suffix == '.zip':
@@ -211,12 +219,18 @@ class TestWPS:
             assert ds.GetGeoTransform() == geo_transform
             ds = None
 
-    def test_010_smoderp2d_capabilities(self):
+    def test_010_soil_texture_hsg20km2(self):
+        ofile = self._run_job_request(
+            Path(__file__).parent / 'request-soil-texture-hsg20km2.xml',
+            exception="Process error: Limit 20km2 na vymeru zajmoveho uzemi prekrocen"
+        )
+
+    def test_011_smoderp2d_capabilities(self):
         processes = self._wps('https://rain1.fsv.cvut.cz:4444/services/wps').processes
         assert len(processes) == 2
         assert any(process.identifier.startswith('smoderp') for process in processes)
 
-    def test_011_smoderp2d_line(self):
+    def test_012_smoderp2d_line(self):
         ofile = self._run_job_request(        
             Path(__file__).parent / 'request-smoderp2d-line.xml',
             '.csv',

--- a/ows_tests/wps/test_wps.py
+++ b/ows_tests/wps/test_wps.py
@@ -7,6 +7,8 @@ from owslib.wps import WebProcessingService, monitorExecution, ComplexDataInput
 from osgeo import ogr, gdal
 
 class TestWPS:
+    url='https://rain1.fsv.cvut.cz/services/wps'
+    # url='http://localhost/services/wps'
     input_data=ComplexDataInput("http://rain.fsv.cvut.cz/geodata/test.gml")
     key="HLGP_ID"
     return_period="N2,N5,N100"
@@ -18,7 +20,7 @@ class TestWPS:
 
     def _wps(self, url=None):
         return WebProcessingService(
-            url if url else 'https://rain1.fsv.cvut.cz/services/wps'
+            url if url else self.url
         )
 
     def _get_filename(self):
@@ -89,7 +91,7 @@ class TestWPS:
     def test_002_d_rain_shp_post(self):
         """Test d-rain-shp (post method)."""
         ofile = self._run_job_request(
-            './tests/wps/request-d-rain-shp.xml',
+            Path(__file__).parent / 'request-d-rain-shp.xml',
             '.zip'
         )
 
@@ -197,7 +199,7 @@ class TestWPS:
                 
     def test_009_soil_texture_hsg(self):
         ofile = self._run_job_request(
-            './tests/wps/request-soil-texture-hsg.xml',
+            Path(__file__).parent / 'request-soil-texture-hsg.xml',
             '.zip'
         )
 
@@ -216,7 +218,7 @@ class TestWPS:
 
     def test_011_smoderp2d_line(self):
         ofile = self._run_job_request(        
-            './tests/wps/request-smoderp2d-line.xml',
+            Path(__file__).parent / 'request-smoderp2d-line.xml',
             '.csv',
             'https://rain1.fsv.cvut.cz:4444/services/wps'
         )

